src/ACEtk/block/details/BaseDistributionDataWithInterpolationType/src/generateBlocks.hpp

```diff
@@ -1,21 +1,27 @@
 void generateBlocks() {
 
-  std::size_t ne = static_cast< std::size_t >( this->XSS( 2 ) );
-  auto data = this->begin() + 1;
-  auto end = data + 2 * ne + 1;
+  std::size_t ne = static_cast< std::size_t >( this->IXSS( 2 ) );
+  auto data = this->iterator( 2 );
+  auto end = this->iterator( 2 + 2 * ne + 1 );
 
   this->values_ = block::details::ColumnData( std::string( this->name() ),
                                               data, end, 2 );
-  for ( unsigned int index = 1; index <= this->values_.N(); ++index ) {
+  for ( unsigned int index = 1; index <= this->numberValues(); ++index ) {
 
     const double value = this->value( 1, index );
     const auto locb = this->distributionLocator( index );
     const auto locator = this->relativeDistributionLocator( index );
-    const auto left = std::next( this->begin(), locator - 1 );
-    const auto right = index == this->numberValues()
-                       ? this->end()
-                       : std::next( this->begin(),
-                         this->relativeDistributionLocator( index + 1 ) - 1 );
+    const auto left = this->iterator( locator );
+    auto right = this->end();
+    for ( auto next = index + 1; next <= this->numberValues(); ++next ) {
```

Comment:
[cleanup suggestion]
I see this linear search repeated in a number of locations. Should it be spun off into its own routine?

[performance]
Should a linear search be used always? Linear search is advantageous for n small (typically n < 100). But that is only due to the contiguity of memory. The contiguity of memory is not present in most of these searches.

Reply:
Yes, probably for later.